remote-dependency-templates/README.md

```diff
@@ -1,11 +1,74 @@
-## Usage
+# Using dependency templates
 - List dependency `<id>` as a remote dependency in `application.yml`, with one of the supported types
   - Populate required `template-config` fields (defined in the README for each type)
 - List any dependency-specific service data under `dependency-data.<id>` in a service's `service.yml`
 
-## Terraform variables //TODO
+# Building dependency templates
+- Create a directory with a unique name. This will be the dependency's `type`.
+- The directory should contain the following three files: requirements.yml, dependency.tf and README.md. Each is explained in detail below. (see existing templates for example)
 
-## Notes
-The following are fields that exosphere passes to every dependency:
-- `app-name`: Name of the application as defined in `application.yml`
+## requirements.yml
+Contains a single field: `required-fields`, which is a list of strings that define the fields that a user is required to populate
+under the `template-config` section of a dependency configuration.
 
+Example:
+```yml
+# requirements.yml
+
+required-fields:
+  - version
+```
+```yml
+# application.yml
+
+<dependency-id>:
+  type: <type>
+  template-config:
+    version: 0.0.1
+```
+
+## dependency.tf
+A terraform file that contains the public interfacing terraform modules for a dependency. The file should be built
+as a [Mustache template](https://github.com/hoisie/mustache). Any configuration information listed as a required field in `requirements.yml` (see above)
+will be rendered into `dependency.tf` using Mustache. The mustache variables must match those listed in `requirements.yml`, and hence in a user's `template-config`.
+Example:
+```tf
+# dependency.tf
+module "example_module" {
+  name = {{dependency-name}}
+}
+```
+```yml
+# requirements.yml
+required-fields:
+ - dependency-name
+```
+```yml
+# application.yml
+dependencies:
+  <dependency-id>:
+    type: example
+    template-config:
+      dependency-name: this-will-be-rendered
+```
```

Comment:
Thoughts on updating the example to

```
<dependency.tf>
<application.yml>
<rendered dependency in the users main.tf>
```
I don't think we need the `requirements.yml`.